### PR TITLE
improve coverage

### PR DESCRIPTION
**Pull request by giovp**
remove coverage from `types.py` and init tests for `decomposition.py`, partially address #254 

Had some issues with passing correct arrays to `SparseCholesky` so could try to improve interface a bit.

### CODE REVIEW

**Comment by giovp** [comment]
I'm not sure why tests are failing, seems pip can't resolve dependencies, shall I unpin up `scikit-sparse`

File: tests/math/decomposition_test.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from typing import Any
+
+import pytest
+
+import jax
+import jax.numpy as jnp
+from jax.experimental import sparse
+
+from ott.math.decomposition import DenseCholeskySolver, SparseCholeskySolver
+
```
Comment (michalk8) [inline tests/math/decomposition_test.py:10]: Please add [this](https://github.com/ott-jax/ott/blob/main/tests/geometry/graph_test.py#L21)

File: pyproject.toml [hunk 2]
```diff
@@ -72,7 +72,8 @@ test = [
     "coverage[toml]",
     "chex",
     "networkx>=2.5",
-    "scikit-learn>=1.0"
+    "scikit-learn>=1.0",
+    "scikit-sparse>=0.4.6"
```
Comment (michalk8) [inline pyproject.toml:76]: This is already handled in `tox`, please remove.

**Review by michalk8 — CHANGES_REQUESTED** [review]
We don't have CHOLMOD on the GPU server (yet), so tests there are disabled (via tox), that's why it fails. On all others, it should be fine.

File: .coveragerc [hunk 3]
```diff
@@ -0,0 +1,2 @@
+[run]
+omit = src/ott/types.py
```
Comment (michalk8) [inline .coveragerc:2]: This is already specified in [pyproject.toml](https://github.com/ott-jax/ott/blob/main/pyproject.toml#L121).

**Comment by marcocuturi** [comment]
@giovp could you take a look maybe? we might want to merge, because we want to increase coverage :)

**Comment by giovp** [comment]
@marcocuturi resolved conflicts, any pointer on what else to be done LMK

**Pull request closed without merging**